docs/config-file/v2.rst

```diff
@@ -0,0 +1,365 @@
+Configuration File V2
+=====================
+
+Read the Docs supports configuring your documentation builds with a YAML file.
+:doc:`The Read the Docs file <index>` must be in the root directory of your project.
+
+Here is an example of how this file looks like:
+
+.. code:: yaml
+
+   # .readthedocs.yml
+
+   version: 2
+
+   python:
+     version: 3.6
+     install: pip
+
+Supported settings
+------------------
+
+version
+~~~~~~~
+
+:Required: ``true``
+
+.. code-block:: yaml
+   
+   version: 2
+
+.. warning::
+   
+   If you don't provide the version, :doc:`v1 <v1>` would be used.
+
+formats
+~~~~~~~
+
+Formats of the documentation to be built.
+
+:Type: ``list``
+:Options: ``htmlzip``, ``pdf``, ``epub``
+:Default: ``[]``
+
+.. code-block:: yaml
+
+   # Default
+   formats: []
+
+.. code-block:: yaml
+
+   # Build PDF & ePub
+   formats:
+     - epub
+     - pdf
+
+.. note::
+
+   You can use the ``all`` keyword to indicate all formats.
+
+   .. code-block:: yaml
+
+      # Build all formats
+      formats: all
+
+python
+~~~~~~
+
+Configuration of the Python environment to be used.
+
+.. code-block:: yaml
+
+   python:
+      version: 3.6
+      requirements: docs/requirements.txt
+      install: pip
+      extra_requirements:
+        - docs
+      system_packages: true
+
+python.version
+``````````````
+
+The Python version (this depends on :ref:`config-file/v2:build.image`).
+
+:Type: ``number``
+:Default: ``3``
+
+python.requirements
+```````````````````
+
+The path to the requirements file from the root of the project.
+
+:Type: ``path``
+:Default: ``null``
+
+If the value is ``null``,
+Read the Docs would try to find a requirements file in your ``docs/`` directory and in the root of your project.
+If you want to completely ignore the requirements file, you can use an empty string ``''``.
```

Comment:
https://github.com/rtfd/readthedocs.org/issues/4523

Reply:
If we are going to change this, *now* is the right moment. Otherwise, when people start using this will be a problem (or something else to maintain in the future).

Reply:
the docs are the last thing to merge so we still have time